modules/openapi-generator/src/test/resources/3_0/oneOfOverlapingTypes.yaml

```diff
@@ -0,0 +1,48 @@
+openapi: 3.0.1
+info:
+  title: fruity
+  version: 0.0.1
+paths:
+  /:
+    get:
+      responses:
+        "200":
+          description: desc
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/fruit"
+components:
+  schemas:
+    fruit:
+      title: fruit
+      oneOf:
+        - $ref: "#/components/schemas/apple"
+        - $ref: "#/components/schemas/banana"
+    appleColor:
+      type: string
+      enum:
+        - yellow
+        - green
+        - red
+    apple:
+      title: apple
+      type: object
+      properties:
+        kind:
+          type: string
+        color:
+          $ref: '#/components/schemas/appleColor'
```

Comment:
Can you make the color property required in both apple and banana?
Some languages treat required and optional variables differently, and making it required makes it required at the composed schema level also.

Reply:
Just to add that the `Fruit` definition seems to be invalid with respect to how `oneOf` is defined.

Consider `{"color":"green"}` which will match both Apple and Banana and result in the violation of the `oneOf` definition (one and only one match)

Reply:
@wing328 `{"color":"green"}` this is the whole point! There is an overlap in a single field but you need other fields to determine the right model.

Reply:
Then those other fields need to be required instead. Otherwise, `{"color":"green"}` will match both.

Reply:
So is the point of adding these tests to show differences in enum values or in the data type of the property?
My understanding was that the data type of the property was the bigger issue.
For example having:
- color be string in one model
- color be integer in another